pubsublite/admin.go

```diff
@@ -155,35 +156,47 @@ func (ac *AdminClient) Topics(ctx context.Context, parent string) *TopicIterator
 }
 
 type createSubscriptionSettings struct {
-	backlogLocation BacklogLocation
+	target SeekTarget
 }
 
 // CreateSubscriptionOption is an option for AdminClient.CreateSubscription.
 type CreateSubscriptionOption interface {
 	apply(*createSubscriptionSettings)
 }
 
-type startingOffset struct {
-	backlogLocation BacklogLocation
+type targetLocation struct {
+	target SeekTarget
 }
 
-func (so startingOffset) apply(settings *createSubscriptionSettings) {
-	settings.backlogLocation = so.backlogLocation
+func (tl targetLocation) apply(settings *createSubscriptionSettings) {
+	settings.target = tl.target
 }
 
 // StartingOffset specifies the offset at which a newly created subscription
 // will start receiving messages.
 func StartingOffset(location BacklogLocation) CreateSubscriptionOption {
-	return startingOffset{location}
+	return targetLocation{location}
+}
+
+// AtTargetLocation specifies the target location within the message backlog
+// that a new subscription should be initialized to.
+//
+// An additional seek request is initiated if the target location is a publish
+// or event time. If the seek fails, the created subscription is not deleted.
+func AtTargetLocation(target SeekTarget) CreateSubscriptionOption {
```

Comment:
Is StartingOffset still needed after this? Should it be deprecated?

Reply:
I added a comment stating that StartingOffset will be removed in the next major revision.